### PR TITLE
INT B-22654

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-22654)

> [!NOTE]  
> This branch contains the code of [B-22653](https://github.com/transcom/mymove/pull/14889) so please ensure that is merged prior to this one once POA'd

## Summary

Here we are setting up the groundwork for a new type of PPM called "Small Package Reimbursement". With the work of [B-22653](https://github.com/transcom/mymove/pull/14889), we now have the database changes and feature flag additions that we need to start work.

This PR specifically focuses on the PPM creation/updating process for the following scenarios:
- customer creating a PPM shipment 
- customer created PPMs should have a default `ppm_type` of `INCENTIVE_BASED` _if they are **NOT** civilian employees_
- customer created PPMs should have a default `ppm_type` of `ACTUAL_EXPENSE` _if they **ARE** civilian employees_
- if the customer edits their orders after PPM creation and changes their pay grade _FROM_ `CIVILIAN_EMPLOYEE` or _TO_ `CIVILIAN_EMPLOYEE`, the `ppm_type` needs to be updated if the PPM is still in `DRAFT` status
- the customer can then edit their PPM freely but should not even know what kind of PPM type they have because we aren't showing that to them - only the SC
- once it gets to the SC, we want to let them choose the PPM type (note: we are removing the previous actual reimbursement block since it is being covered in the type selection)
- SC should be able to edit the PPM types (civilians should have a disabled `Incentive-based` option)
- should be able to download the AOA packet and see the SPR values on that PDF

### How to test

1. Make sure you turn on the `FEATURE_FLAG_PPM_SPR` flag in your `envrc`
2. Fire up your `client` & `server`
3. Go through setting up a PPM as a service member (not a civilian), nothing should be changed here and shouldn't get any errors
4. Submit the PPM
5. Go in as SC and review the PPM - you should now see an option to designate the PPM as other options (`Actual Expense` & `Small Package`)
6. Confirm that the `Incentive-based` option is checked by default
7. Play around with it, you should see some labels change when you select `Small Package Reimbursement`
8. Make sure you can successfully update any PPM of any type as SC
9. Make sure you can do the same as TOO (note: TOOs do not see the PPM type designator)
10. Set up a PPM that gets an advance so you can download the AOA packet
11. Review the AOA packet and make sure you can see the specific Small Package changes on the SSW 
12. Go through this same process with an actual expense reimbursement and make sure nothing is broken
13. Go through the same process with incentive-based and make sure nothing is broken
14. Now repeat these steps with the pay grade `Civilian Employee` and confirm that SCs cannot select `Incentive-based` for civilians
15. Turn the `FEATURE_FLAG_PPM_SPR` off and make sure all is good and no small package things (on the UI side) cannot be seen

## Screenshots
Non-civilian incentive-based
Screenshot 2025-02-27 at 10 35 28 AM

Non-civilian actual expense
Screenshot 2025-02-27 at 10 35 09 AM

Non-civilian small package
Screenshot 2025-02-27 at 10 35 14 AM

Civilian actual expense
Screenshot 2025-02-27 at 11 08 11 AM

Civilian small package
Screenshot 2025-02-27 at 11 08 16 AM

Screenshot 2025-03-03 at 1 38 16 PM
Screenshot 2025-03-03 at 1 38 22 PM
Screenshot 2025-03-03 at 1 38 27 PM
Screenshot 2025-03-03 at 1 38 40 PM

### CODE REVIEW

**Review by cameroncaci — COMMENTED** [review]
I broke it. These are the steps you can take to submit an incentive based PPM as a civilian employee

1. As a civilian customer, create your order but DO NOT set your pay grade as employee. Make yourself an "OTHER" "E-4" for example
2. Submit your PPM
3. At this point, the PPM defaults to "Incentive-based"
4. As SC, review the orders, correct the pay grade setting it back to civilian
5. As SC, don't change the PPM type to actual expense report, just leave it as default
6. As SC, submit the move. And blammo, you have now submitted a civilian incentive based PPM which can't be changed

Screenshot 2025-03-04 at 9 44 30 AM

**Comment by cameroncaci** [comment]
As SC reviewing submitted PPM expenses, we don't label the small PPM packet. is that something we want included with this PR?
Pasted Graphic

**Comment by danieljordan-caci** [comment]
> As SC reviewing submitted PPM expenses, we don't label the small PPM packet. is that something we want included with this PR? Pasted Graphic

Prob wouldn't hurt - I'll add it

**Comment by danieljordan-caci** [comment]
> I broke it. These are the steps you can take to submit an incentive based PPM as a civilian employee
> 
> 1. As a civilian customer, create your order but DO NOT set your pay grade as employee. Make yourself an "OTHER" "E-4" for example
> 2. Submit your PPM
> 3. At this point, the PPM defaults to "Incentive-based"
> 4. As SC, review the orders, correct the pay grade setting it back to civilian
> 5. As SC, don't change the PPM type to actual expense report, just leave it as default
> 6. As SC, submit the move. And blammo, you have now submitted a civilian incentive based PPM which can't be changed
> 
> Screenshot 2025-03-04 at 9 44 30 AM

Valid edge case - handled this for customer but looks like should address as office user as well when editing pay grade

**Comment by cameroncaci** [comment]
> Prob wouldn't hurt - I'll add it

Whoops, forgot to include the fileline src/components/Office/PPM/PPMHeaderSummary/PPMHeaderSummary.jsx:92

**Comment by cameroncaci** [comment]
I did also find this font sizing issue with the SSW, not sure if this was preexisting or not
image

**Comment by danieljordan-caci** [comment]
> I did also find this font sizing issue with the SSW, not sure if this was preexisting or not - nothing to panic over but thought it was worth mentioning image

Dude I tried everything I could to fix that. It's random too and sometimes happens and sometimes doesn't. Explicitly setting font size in the PDF doesn't seem to work. I have no idea what's going on here.

**Comment by cameroncaci** [comment]
> > I did also find this font sizing issue with the SSW, not sure if this was preexisting or not - nothing to panic over but thought it was worth mentioning image
> 
> Dude I tried everything I could to fix that. It's random too and sometimes happens and sometimes doesn't. Explicitly setting font size in the PDF doesn't seem to work. I have no idea what's going on here.

@antgmann help, we need your 🧠. Seen this before?

**Comment by antgmann** [comment]
> @antgmann help, we need your 🧠. Seen this before?

PDFCPU doesn't like to respect the font size limit set in the pdf template 😔  this causes the above to happen and seem really inconsistent. Fortunately, I've been able to notice that once you get it to the right size and export it, it should stay that way. Which leads me to ask:
Are the font issues only showing on fields that have been edited as part of this work? If so, then it's likely because the size of the form field is too large. PDFCPU _seems to_ automatically change font size to match what it thinks will best fit the dimensions of the form field, and it's often wrong. If you re-size the fields to be smaller, export, try again, repeat, you should be able to get a more consistent result. See: https://github.com/transcom/mymove/pull/12280 commit https://github.com/transcom/mymove/pull/12280/commits/9000a0b8a618765bfaa3201b607588335e1a4a22 . 

It's not nice to deal with, but it should work for you.

**Comment by danieljordan-caci** [comment]
> > @antgmann help, we need your 🧠. Seen this before?
> 
> PDFCPU doesn't like to respect the font size limit set in the pdf template 😔 this causes the above to happen and seem really inconsistent. Fortunately, I've been able to notice that once you get it to the right size and export it, it should stay that way. Which leads me to ask: Are the font issues only showing on fields that have been edited as part of this work? If so, then it's likely because the size of the form field is too large. PDFCPU _seems to_ automatically change font size to match what it thinks will best fit the dimensions of the form field, and it's often wrong. If you re-size the fields to be smaller, export, try again, repeat, you should be able to get a more consistent result. Also, be sure to only edit the fields that are having the issue. While my above theory seems to be mostly correct, it's not as scientific as I'd like. See: #12280 commit [9000a0b](https://github.com/transcom/mymove/commit/9000a0b8a618765bfaa3201b607588335e1a4a22) .
> 
> It's not nice to deal with, but it should work for you.

Thanks brutha - I'll give this a crack

**Comment by danieljordan-caci** [comment]
> As SC reviewing submitted PPM expenses, we don't label the small PPM packet. is that something we want included with this PR? Pasted Graphic

Actually I want to take this back because this is a separate component and the closeout portion of this work is coming next so it would be more appropriate to address it there. This BL specifically covers from customer submission to SC submitting move details.

**Comment by danieljordan-caci** [comment]
> I broke it. These are the steps you can take to submit an incentive based PPM as a civilian employee
> 
> 1. As a civilian customer, create your order but DO NOT set your pay grade as employee. Make yourself an "OTHER" "E-4" for example
> 2. Submit your PPM
> 3. At this point, the PPM defaults to "Incentive-based"
> 4. As SC, review the orders, correct the pay grade setting it back to civilian
> 5. As SC, don't change the PPM type to actual expense report, just leave it as default
> 6. As SC, submit the move. And blammo, you have now submitted a civilian incentive based PPM which can't be changed
> 
> Screenshot 2025-03-04 at 9 44 30 AM

Got you fixed up in [this commit](https://github.com/transcom/mymove/pull/14934/commits/a85f92d2d35e7a142252d3d50e449c5221bdafc0)

Re-requesting review

**Review by cameroncaci — DISMISSED** [review]
Good edge case fix and test coverage, agree with the closeout component being in the next backlog. G2g 👍🏻

**Comment by traskowskycaci** [comment]
> Only failing test is int specific primeapiv3

This one should be fixed as of last night

File: src/components/Office/ShipmentForm/ShipmentForm.jsx [hunk 1]
```diff
@@ -107,17 +107,23 @@ const ShipmentForm = (props) => {
...
+
   useEffect(() => {
     const fetchData = async () => {
       setIsTertiaryAddressEnabled(await isBooleanFlagEnabled('third_address_available'));
     };
     fetchData();
   }, []);
+  useEffect(() => {
+    const fetchData = async () => {
+      setPpmSprFF(await isBooleanFlagEnabled('ppm_spr'));
```
Comment (ryan-mchugh) [inline src/components/Office/ShipmentForm/ShipmentForm.jsx:123]: Can you use the constant as you did in other files?

image

File: src/components/Office/ShipmentForm/ShipmentForm.jsx [hunk 2]
```diff
@@ -298,7 +302,7 @@ const ShipmentForm = (props) => {
             closeoutOffice: move.closeoutOffice,
           },
     );
-    if (isCreatePage && serviceMember?.grade === 'CIVILIAN_EMPLOYEE')
+    if (isCreatePage && mtoShipment.ppmShipment?.ppmType === PPM_TYPES.ACTUAL_EXPENSE)
       initialValues.isActualExpenseReimbursement = 'true';
```
Comment (ryan-mchugh) [inline src/components/Office/ShipmentForm/ShipmentForm.jsx:306]: can remove this line altogether since data source is now ppmType===`PPM_TYPES.ACTUAL_EXPENSE`

Comment (danieljordan-caci) [reply src/components/Office/ShipmentForm/ShipmentForm.jsx]: Fixed it up in [this commit](https://github.com/transcom/mymove/pull/14934/commits/76b1c9ee87890d83289d3c87db181bd530d0b3e4)

File: src/components/Office/ShipmentForm/ShipmentForm.jsx (hunk 1 above)
Comment (danieljordan-caci) [reply src/components/Office/ShipmentForm/ShipmentForm.jsx]: Here ya go [here](https://github.com/transcom/mymove/pull/14934/commits/76b1c9ee87890d83289d3c87db181bd530d0b3e4)

**Review by cameroncaci — APPROVED** [review]
Reapproving

**Comment by ryan-mchugh** [comment]
Local Testing
image
image

Field offset and partially hidden for actual obligations (specifically when downloaded and viewed in preview, not chrome)
image
image
image


In chrome, it's good
image
image

**Comment by ryan-mchugh** [comment]
A lot of shipment worksheet test failures


image

**Comment by danieljordan-caci** [comment]
Re-requesting review from @cameroncaci , @TevinAdams & @ryan-mchugh 

Had to resolve some merge conflicts

**Comment by danieljordan-caci** [comment]
There are currently failing SSW tests on INT - these are not relevant to this BL. My `main` PR has all SSW tests passing

**Review by cameroncaci — APPROVED** [review]
reapproving

**Comment by ryan-mchugh** [comment]
`shipment_summary_worksheet.test.go` is failing in INT, but have determined that it passes on the main branch and was already failing in integration prior to this work. Approving ✅ 

image

**Review by ryan-mchugh — APPROVED** [review]
